include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,405 @@
+
+    static constexpr size_t __actual_padding_stride = detail::__get_actual_padding_stride<extents_type, padding_stride>();
+
+    using __inner_extents_type = typename detail::__inner_extents_left<extents_type, __actual_padding_stride>::__type;
+    using __unpadded_extent_type = typename detail::__unpadded_extent_type_impl<extents_type>::__type;
+    using __inner_mapping_type = layout_left::template mapping<__inner_extents_type>;
+
+    __inner_mapping_type __inner_mapping;
+    __unpadded_extent_type __unpadded_extent;
+
+public:
+#if !MDSPAN_HAS_CXX_20
+    MDSPAN_INLINE_FUNCTION_DEFAULTED
+    constexpr mapping()
+        : mapping(extents_type{})
+    {}
+#else
+    MDSPAN_INLINE_FUNCTION_DEFAULTED
+    constexpr mapping()
+      requires(__actual_padding_stride != dynamic_extent) = default;
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr mapping()
+      requires(__actual_padding_stride == dynamic_extent)
+        : mapping(extents_type{})
+    {}
+#endif
+
+    MDSPAN_INLINE_FUNCTION_DEFAULTED constexpr mapping(const mapping&) noexcept = default;
+    MDSPAN_INLINE_FUNCTION_DEFAULTED mapping& operator=(const mapping&) noexcept = default;
+
+    MDSPAN_INLINE_FUNCTION constexpr
+    mapping(const extents_type& __ext)
+      : __inner_mapping(detail::__inner_extents_left<extents_type, __actual_padding_stride>::template __construct<padding_stride>(__ext)),
+        __unpadded_extent(detail::__unpadded_extent_type_impl<extents_type>::__construct(__ext))
+    {}
+
+    template <class _Size>
+    // TODO: constraints
+    constexpr mapping(const extents_type &__ext, _Size __padding_value)
+        : __inner_mapping(detail::__inner_extents_left<extents_type, __actual_padding_stride>::template __construct<padding_stride>(__ext, static_cast<index_type>(__padding_value))),
+          __unpadded_extent(detail::__unpadded_extent_type_impl<extents_type>::__construct(__ext))
+    {
+      assert((padding_stride == dynamic_extent) || (padding_stride == static_cast<index_type>(__padding_value)));
+    }
+
+    template <class _OtherExtents>
+    constexpr MDSPAN_CONDITIONAL_EXPLICIT((!is_convertible_v<_OtherExtents, extents_type>))
+        mapping(const layout_left::mapping<_OtherExtents> &__other_mapping)
+        : __inner_mapping(detail::__inner_extents_left<extents_type, __actual_padding_stride>::template __construct_other<padding_stride>(__other_mapping.extents(), __other_mapping.stride(0))),
+          __unpadded_extent(detail::__unpadded_extent_type_impl<extents_type>::__construct(__other_mapping.extents()))
+    {
+    }
+
+    template <class _OtherExtents>
+    constexpr MDSPAN_CONDITIONAL_EXPLICIT((extents_type::rank() > 0))
+        mapping(const layout_stride::mapping<_OtherExtents> &__other_mapping)
+        : __inner_mapping(detail::__inner_extents_left<extents_type, __actual_padding_stride>::template __construct_other<padding_stride>(__other_mapping.extents(), __other_mapping.stride(0))),
```

Comment:
same here: those strides need to be `stride(1)` or? because that is what the extra thing is we do in `layout_left_padded` vs `layout_left`.

Reply:
Ah checked the paper, also says `stride(1)`

Reply:
yeah looks like needs to be stride(1). fixed